File: 프로그래머스/BFS.DFS/네트워크.cpp
#include <string>
#include <vector>

using namespace std;
int visit[200];
int count = 0;
void dfs(vector<vector<int>> computers,int a){
    
    if(visit[a]) return;
    visit[a] = 1;
    for(int i = 0; i < computers[a].size(); i++){
        if(computers[a][i] == 1)dfs(computers, i);
    }
    
    
}


int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    for(int i = 0 ; i < computers.size(); i++){
        if(visit[i] == 0){
            dfs(computers, i);
            count++;
        }
        
    }
    
    answer = count;
    
    
    
    
    return answer;
}

File: 프로그래머스_python/분류별/DFS_BFS/네트워크.py
def dfs(graph,visit,x):
    
    start = x
    visit[x] = 1
    
    for i in graph[x]:
        if visit[i] != 1 :
            dfs(graph,visit,i)

    


def solution(n, computers):
    answer = 0
    
    graph = [[] for _ in range(n)]
    visit = [0] * n
    
    for i in range(len(computers)):
        for j in range(len(computers[i])):
            if i == j:
                continue
            
            elif computers[i][j] == 1:
                graph[i].append(j)
    
    for i in range(len(graph)):
        if visit[i] == 1:
            continue
        
        dfs(graph,visit,i)
        answer += 1
        
    
            
    
    
    
    
    
    
    return answer
